.jenkins/pytorch/win-test.sh

```diff
@@ -72,13 +72,13 @@ run_tests() {
           export PYTORCH_COLLECT_COVERAGE=1
           export COVERAGE_RCFILE=$PWD/.coveragerc # coverage config file needed for plug-ins and settings to work
         fi
-        if [[ "${JOB_BASE_NAME}" == *-test1 ]]; then
+        if [[ "${JOB_BASE_NAME}" == *-test1 || ( $NUM_TEST_SHARDS -gt 1 && "${TEST_CONFIG}" == 1 )]]; then
             "$SCRIPT_HELPERS_DIR"/test_python_first_shard.bat "$DETERMINE_FROM"
             "$SCRIPT_HELPERS_DIR"/test_libtorch.bat
             if [[ "${USE_CUDA}" == "1" ]]; then
               "$SCRIPT_HELPERS_DIR"/test_python_jit_legacy.bat "$DETERMINE_FROM"
             fi
-        elif [[ "${JOB_BASE_NAME}" == *-test2 ]]; then
+        elif [[ "${JOB_BASE_NAME}" == *-test2 || ( $NUM_TEST_SHARDS -gt 1 && "${TEST_CONFIG}" == 2 )]]; then
```

Comment:
not related to this PR, is it a good idea to generalize NUM_TEST_SHARDS to handle more than `2`?

Reply:
I think we'd potentially want that in the future, thought right now it's not something we can do